great_expectations/core/usage_statistics/anonymizers/profiler_run_anonymizer.py

```diff
@@ -0,0 +1,196 @@
+            clean_falsy=True,
+            inplace=True,
+        )
+
+        return anonymized_profiler_run_properties_dict
+
+    def _anonymize_rules(self, rules: Dict[str, dict]) -> List[dict]:
+        anonymized_rules: List[dict] = []
+
+        for name, rule in rules.items():
+            anonymized_rule: dict = self._anonymize_rule(name, rule)
+            anonymized_rules.append(anonymized_rule)
+            logger.debug("Anonymized rule %s", name)
+
+        return anonymized_rules
+
+    def _anonymize_rule(self, name: str, rule: dict) -> dict:
+        anonymized_rule: dict = {}
+        anonymized_rule["anonymized_name"] = self.anonymize(name)
+
+        domain_builder: Optional[dict] = rule.get("domain_builder")
+        if domain_builder is not None:
+            anonymized_rule[
+                "anonymized_domain_builder"
+            ] = self._anonymize_domain_builder(domain_builder)
+
+        parameter_builders: List[dict] = rule.get("parameter_builders", [])
+        anonymized_rule[
+            "anonymized_parameter_builders"
+        ] = self._anonymize_parameter_builders(parameter_builders)
+
+        expectation_configuration_builders: List[dict] = rule.get(
+            "expectation_configuration_builders", []
+        )
+        anonymized_rule[
+            "anonymized_expectation_configuration_builders"
+        ] = self._anonymize_expectation_configuration_builders(
+            expectation_configuration_builders
+        )
+
+        return anonymized_rule
+
+    def _anonymize_domain_builder(self, domain_builder: dict) -> dict:
+        anonymized_domain_builder: dict = {}
+        anonymized_domain_builder["class_name"] = domain_builder.get("class_name")
+
+        batch_request: Optional[dict] = domain_builder.get("batch_request")
+        if batch_request:
+            anonymized_domain_builder[
+                "anonymized_batch_request"
+            ] = self._batch_request_anonymizer.anonymize_batch_request(
+                *(), **batch_request
+            )
+            logger.debug("Anonymized batch request in DomainBuilder")
+
+        return anonymized_domain_builder
+
+    def _anonymize_parameter_builders(
+        self, parameter_builders: List[dict]
+    ) -> List[dict]:
+        anonymized_parameter_builders: List[dict] = []
+
+        for parameter_builder in parameter_builders:
+            anonymized_parameter_builder: dict = self._anonymize_parameter_builder(
+                parameter_builder
+            )
+            anonymized_parameter_builders.append(anonymized_parameter_builder)
+
+        return anonymized_parameter_builders
+
+    def _anonymize_parameter_builder(self, parameter_builder: dict) -> dict:
+        anonymized_parameter_builder: dict = {}
+
+        anonymized_parameter_builder["anonymized_name"] = self.anonymize(
+            parameter_builder.get("name")
+        )
+
+        anonymized_parameter_builder["class_name"] = parameter_builder.get("class_name")
```

Comment:
This class name will also need to be anonymized if it is not one of the GE provided class names. See note above.